base.go

```diff
@@ -410,18 +412,23 @@ func filterError(e *Error) error {
 	return e
 }
 
-// NonStrict wraps a value v so that it can be unmarshaled from JSON without
-// checking for unknown fields. The v provided must itself be a valid argument
-// to json.Unmarshal.
+// StrictFielder is an optional interface that can be implemented by a type to
+// reject unknown fields when unmarshaling from JSON.  If a type does not
+// implement this interface, unknown fields are ignored.
+type StrictFielder interface {
+	StrictFields()
+}
+
+// StrictFields wraps a value v so that it can be unmarshaled as a parameter
+// value from JSON without checking for unknown fields.
 //
-// This can be used to unmarshal request parameters with unknown fields, for
-// example:
+// For example:
 //
 //       var obj RequestType
-//       err := req.UnmarshalParams(jrpc2.NonStrict(&obj))
+//       err := req.UnmarshalParams(jrpc2.StrictFields(&obj))
 //
-func NonStrict(v interface{}) json.Unmarshaler { return &nonStrict{v: v} }
+func StrictFields(v interface{}) interface{} { return &strict{v: v} }
 
-type nonStrict struct{ v interface{} }
+type strict struct{ v interface{} }
 
-func (n nonStrict) UnmarshalJSON(data []byte) error { return json.Unmarshal(data, n.v) }
+func (strict) StrictFields() {}
```

Comment:
I don't expect `StrictFields` to conflicts, but I'm thinking `DisallowUnknownFields()` would align it better with the nomenclature of `encoding/json.Decoder`?

I guess it would be a bad practice to have a type which is used for (un)marshaling data _and_ implementing some complex logic at the same time anyway, so the naming conflict is probably highly unlikely.

Reply:
I didn't do that because the name felt clunky, but now that you say it explicitly I think you're probably right. Not that I think a name conflict is likely, but it would be more self-documenting.